src/Evolution/DiscontinuousGalerkin/ExponentialFilter.hpp

```diff
@@ -0,0 +1,173 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
+
+#include "DataStructures/Matrix.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Filtering.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/StaticCache.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Filters {
```

Comment:
Is any of this specific to evolutions? Then wrap in the `evolution` namespace. If not, better move this out of the `Evolution` directory into `NumericalAlgorithms`

Reply:
Is filtering ever used in elliptic problems? If not, then I would say that this is evolution specific.

Reply:
The SpEC elliptic solver does some filtering, yes.

Reply:
I prefer to defer moving code until it is known that the SpECTRE elliptic solver will use filtering.  It may be that SpEC needed it to handle overlapping subdomains.

Reply:
You‘re moving this code into new files, essentially factoring out the part that is independent of `Evolution`. It doesn‘t even depend on the parallel architecture anymore, so I think the Filter and the Tags should really go into`NumericalAlgorithms`, if used by the elliptic solver or not

Reply:
@kidder I talked to @hpfeiffer about filtering in the SpEC elliptic solver. It uses filtering (i) to discard modes from a spherical harmonics decomposition (probably irrelevant for us) and (ii) to damp high modes in the outer, compactified spherical shell as part of preconditioning (@hpfeiffer please correct me if I’m getting anything wrong). We may not need filtering in the end, but we’ll certainly experiment with it and it’s possible we’ll end up with something like (ii). Preconditioning is hard and filtering is one tool we’ll use to try to make it better :)

Reply:
Discussion with @wthrowe @kidder @nilsdeppe: `NumericalAlgorithms/LinearOperators` is a good location